storybookjs/storybook
# Scripts: Improve check task to allow for running with linking

## What I did

I was annoyed by the fact I could only run `nr task check` with `--no-link`.

So by calling `tsc`'s node apis and manually filtering diagnostic messages to only those coming from the `cwd`, I was able to get exactly this!

This also opens up the option of adding different "check" actions, such as [#knip](https://github.com/storybookjs/storybook/pull/20572) maybe.

I was also considering some check actions like https://www.npmjs.com/package/check-peer-dependencies or https://github.com/rwjblue/validate-peer-dependencies

The CI still runs the `check` task with `--no-link`.

## How to test

- checkout this branch
- run `yarn task check -s compile`
- expect some packages to get skipped, this is expected: `vue` & `svelte` use some custom `tsc` clone, and I didn't want to invest time there.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`
